staging/src/k8s.io/apimachinery/pkg/util/strategicpatch/patch.go

```diff
@@ -837,6 +837,17 @@ func StrategicMergeMapPatch(original, patch JSONMap, dataStruct interface{}) (JS
 	if err != nil {
 		return nil, err
 	}
+
+	// We need the go struct tags `patchMergeKey` and `patchStrategy` for fields that support a strategic merge patch.
+	// For native resources, we can easily figure out these tags since we know the fields.
+
+	// Because custom resources are decoded as Unstructured and because we're missing the metadata about how to handle
+	// each field in a strategic merge patch, we can't find the go struct tags. Hence, we can't easily  do a strategic merge
+	// for custom resources. So we should fail fast and return an error.
+	if t.Name() == "Unstructured" {
```

Comment:
would `_, ok := dataStruct.(Unstructured); ok` be the same?

Reply:
`_, ok := dataStruct.(*unstructured.Unstructured); ok` would be the same (with the pointer). :+1: 

updated.